packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1168,6 +1169,7 @@ abstract class Element implements BuildContext {
 
   Map<Type, InheritedElement> _inheritedWidgets;
   Set<InheritedElement> _dependencies;
+  Set<Type> _unsatisfiedDependencies;
```

Comment:
_inheritedWidgets is shared across large parts of the tree. If we used it for this, we'd be rebuilding many parts of the tree when a new inherited widget came in.